pkg/build/builder/transient_mounts.go

```diff
@@ -0,0 +1,106 @@
+package builder
+
+import (
+	"fmt"
+	"strings"
+)
+
+// TransientMountOptions are options that change how a transient mount is setup
+type TransientMountOptions struct {
+	NoDev    bool
+	NoExec   bool
+	NoSuid   bool
+	ReadOnly *bool
+}
+
+//TransientMount is a structure to hold the data needed to construct a transient mount
+type TransientMount struct {
+	Source      string
+	Destination string
+	Options     TransientMountOptions
+}
+
+// returns the string representation of a transient mount
+// format: source:destiation:option1,option2,...
+func (tm TransientMount) String() string {
+	var options []string
+
+	if tm.Options.NoDev {
+		options = append(options, "nodev")
+	}
+	if tm.Options.NoExec {
+		options = append(options, "noexec")
+	}
+	if tm.Options.NoSuid {
+		options = append(options, "nosuid")
+	}
+	if tm.Options.ReadOnly != nil {
+		if *tm.Options.ReadOnly {
+			options = append(options, "ro")
+		} else {
+			options = append(options, "rw")
+		}
+	}
+
+	var mount = []string{tm.Source, tm.Destination}
+	if len(options) != 0 {
+		mount = append(mount, strings.Join(options, ","))
+	}
+
+	return strings.Join(mount, ":")
+
+}
+
+// TransientMounts is a map of transient mounts with the destination as the key
+// so we can detect duplicates quickly as we append new ones
+type TransientMounts map[string]TransientMount
+
+// appends a transient mount to the map and returns an error if a duplicate
+// destination is detected
+func (t TransientMounts) append(mount TransientMount) error {
```

Comment:
might be the first time I've seen in practice a non-struct ptr used for this .... cool :-)

Reply:
btw, the map ref is a ptr under the cover iirc correctly, hence you need nil checks when checking map variables, re: related to my comment above with the funcs off of the `TransientMount` struct

Reply:
> might be the first time I've seen in practice a non-struct ptr used for this .... cool :-)

I'm not going to say that this is common practice, or even the correct thing to do, but it worked :)

Reply:
> btw, the map ref is a ptr under the cover iirc correctly, hence you need nil checks when checking map variables, re: related to my comment above with the funcs off of the `TransientMount` struct

I believe that `ok := t[mount.Destination];` equates to a `nil` check, as it returns `false` if the key does not exist. Beyond that we are really only doing a few operations here:
 - Checking for duplicate keys
 - Adding a key/value pair
 - Outputting the map as a slice of strings constructed from the transient mount that is the value
If there is a specific place you think that I should be checking for `nil` let me know and I will add it.

Reply:
OK, then I'll make an official "nit" req to change it to a pointer ref

Reply:
yeah I'm good with this one ... go ahead and mark this one resolved ... but the other thread needs to be "unresolved"